sentry-android-core/src/main/java/io/sentry/android/core/internal/tombstone/TombstoneParser.java

```diff
@@ -0,0 +1,195 @@
+    excTypeValueMap.put("SIGILL", "IllegalInstruction");
+    excTypeValueMap.put("SIGTRAP", "Trap");
+    excTypeValueMap.put("SIGABRT", "Abort");
+    excTypeValueMap.put("SIGBUS", "BusError");
+    excTypeValueMap.put("SIGFPE", "FloatingPointException");
+    excTypeValueMap.put("SIGSEGV", "Segfault");
+  }
+
+  public SentryEvent parse() throws IOException {
+    TombstoneProtos.Tombstone tombstone = TombstoneProtos.Tombstone.parseFrom(tombstoneStream);
+
+    SentryEvent event = new SentryEvent();
+    event.setLevel(SentryLevel.FATAL);
+
+    // must use the "native" platform because otherwise the stack-trace wouldn't be correctly parsed
+    event.setPlatform("native");
+
+    event.setMessage(constructMessage(tombstone));
+    event.setDebugMeta(createDebugMeta(tombstone));
+    event.setExceptions(createException(tombstone));
+    assert event.getExceptions() != null;
+    event.setThreads(createThreads(tombstone, event.getExceptions().get(0)));
+
+    return event;
+  }
+
+  @NonNull
+  private List<SentryThread> createThreads(
+      TombstoneProtos.Tombstone tombstone, SentryException exc) {
+    List<SentryThread> threads = new ArrayList<>();
+    for (Map.Entry<Integer, TombstoneProtos.Thread> threadEntry :
+        tombstone.getThreadsMap().entrySet()) {
+
+      SentryThread thread = new SentryThread();
+      thread.setId(Long.valueOf(threadEntry.getKey()));
+      thread.setName(threadEntry.getValue().getName());
+
+      SentryStackTrace stacktrace = createStackTrace(threadEntry);
+      thread.setStacktrace(stacktrace);
+      if (tombstone.getTid() == threadEntry.getValue().getId()) {
+        thread.setCrashed(true);
+        // even though we refer to the thread_id from the exception,
+        // the backend currently requires a stack-trace in exception
+        exc.setStacktrace(stacktrace);
+      }
+      threads.add(thread);
+    }
+
+    return threads;
+  }
+
+  @NonNull
+  private static SentryStackTrace createStackTrace(
+      Map.Entry<Integer, TombstoneProtos.Thread> threadEntry) {
+    List<SentryStackFrame> frames = new ArrayList<>();
+
+    for (TombstoneProtos.BacktraceFrame frame : threadEntry.getValue().getCurrentBacktraceList()) {
+      SentryStackFrame stackFrame = new SentryStackFrame();
+      stackFrame.setPackage(frame.getFileName());
+      stackFrame.setFunction(frame.getFunctionName());
+      stackFrame.setInstructionAddr(String.format("0x%x", frame.getPc()));
+      frames.add(0, stackFrame);
+    }
+
+    SentryStackTrace stacktrace = new SentryStackTrace();
+    stacktrace.setFrames(frames);
+
+    Map<String, Object> unknown = new HashMap<>();
+    // `libunwindstack` used for tombstones already applies instruction address adjustment:
+    // https://android.googlesource.com/platform/system/unwinding/+/refs/heads/main/libunwindstack/Regs.cpp#175
+    // prevent "processing" from doing it again.
+    unknown.put("instruction_addr_adjustment", "none");
+    stacktrace.setUnknown(unknown);
```

Comment:
`unknown` is mainly used for hybrid SDKs or when the Java SDK is used as a sending mechanism to ensure no data is lost due to parsing. In this case I'd favor extending the `SentryStackTrace` class with a `private @Nullable String instructionAddressAdjustment` field instead.